docs/buildpacks.md

```diff
@@ -0,0 +1,94 @@
+# Buildpacks
+
+The Buildpack, ClusterBuildpack, and ClusterStore resources are a repository of
+[buildpacks](http://buildpacks.io/) packaged in
+[buildpackages](https://buildpacks.io/docs/buildpack-author-guide/package-a-buildpack/)
+that can be used by kpack to build OCI images.
+
+These objects will be referenced by a [builder](builders.md) resource.
+
+### <a id='buildpack'></a>Buildpack Configuration
+
+Buildpack is a namespaced resource that represents an external buildpackage.
+
+```yaml
+apiVersion: kpack.io/v1alpha2
+kind: Buildpack
+metadata:
+  name: sample-buildpack
+spec:
+  serviceAccountName: sample-sa
+  source:
+    # image: gcr.io/paketo-buildpacks/java
+    # image: gcr.io/paketo-buildpacks/java:8.9.0
+    image: gcr.io/paketo-buildpacks/java@sha256:fc1c6fba46b582f63b13490b89e50e93c95ce08142a8737f4a6b70c826c995de
+```
+
+* `serviceAccountName`: A service account with the sercrets needed to pull the buildpack image.
+* `source.image`: The url of the buildpackage.
+
+### <a id='cluster-buildpack'></a>Cluster Buildpack Configuration
+
+ClusterBuildpack is the cluster scoped version of Buildpack
+
+```yaml
+apiVersion: kpack.io/v1alpha2
+kind: ClusterBuildpack
+metadata:
+  name: sample-cluster-buildpack
+spec:
+  serviceAccountRef:
+    name: sample-sa
+    namespace: sample-namespace
+  source:
+    # image: gcr.io/paketo-buildpacks/java
+    # image: gcr.io/paketo-buildpacks/java:8.9.0
+    image: gcr.io/paketo-buildpacks/java@sha256:fc1c6fba46b582f63b13490b89e50e93c95ce08142a8737f4a6b70c826c995de
+```
+
+* `serviceAccountRef`: An object reference to a service account in any namespace. The object reference must contain `name` and `namespace`.
+* `source.image`: The url of the buildpackage.
+
+
+### <a id='cluster-store'></a>Cluster Store Configuration
+
+ClusterStore is a cluster scoped resource that can reference multiple buildpackages.
+
+Corresponding `kp` cli command docs [here](https://github.com/vmware-tanzu/kpack-cli/blob/main/docs/kp_clusterstore.md).
+
+```yaml
+apiVersion: kpack.io/v1alpha2
+kind: ClusterStore
+metadata:
+  name: sample-cluster-store
+spec:
+ serviceAccountRef:
+    name: sample-sa
+    namespace: sample-namespace
+  sources:
+  - image: gcr.io/cf-build-service-public/node-engine-buildpackage@sha256:95ff756f0ef0e026440a8523f4bab02fd8b45dc1a8a3a7ba063cefdba5cb9493
+  - image: gcr.io/cf-build-service-public/npm-buildpackage@sha256:5058ceb9a562ec647ea5a41008b0d11e32a56e13e8c9ec20c4db63d220373e33
+  - image: gcr.io/paketo-buildpacks/builder:base
+```
+
+* `serviceAccountRef`: An object reference to a service account in any namespace. The object reference must contain `name` and `namespace`.
+* `sources`:  List of buildpackage images to make available in the ClusterStore. Each image is an object with the key image.
+
+> Note: ClusterBuilders will also work with a prebuilt builder image if a builpack is not available in a buildpackage.
```

Comment:
Is this true for Builder resources as well?

Reply:
This was originally copied over from `store.md`, but it did lead me to a fascinating and terrifying conclusion:

I tried this out and it's both valid and works

```yaml
apiVersion: kpack.io/v1alpha2
kind: Buildpack
metadata:
  name: my-buildpack
spec:
  source:
    image: index.docker.io/paketobuildpacks/builder-jammy-tiny
```

Reply:
Updated blurb in 51a0f1752390b3351921fbd0507fee9cd7612a8b